src/etc/inc/captiveportal.inc

```diff
@@ -2841,4 +2899,24 @@ function captiveportal_isip_logged($clientip) {
 		return $cpentry;
 	}
 }
+
+// Return true if the CARP status of at least one interface of a captive portal zone is in backup mode
+function captiveportal_ha_is_node_in_backup_mode() {
+	global $config, $cpzone;
+
+	$cpinterfaces = explode(",", $config['captiveportal'][$cpzone]['interface']);
+
+	if (is_array($config['virtualip']['vip'])) {
+		foreach ($cpinterfaces as $interface) {
+			foreach ($config['virtualip']['vip'] as $vip) {
+				if ($vip['interface'] == $interface && $vip['mode'] == "carp") {
```

Comment:
The first test here needs its own set of parenthesis

Reply:
Do you mean something like ```if (($vip['interface'] == $interface) && $vip['mode'] == "carp") {``` ?

Reply:
They both should be inside their own sets of parenthesis actually...

`if (($vip['interface'] == $interface) && ($vip['mode'] == "carp")) {`